File: backend/src/main/java/tom/projects/monkeylog/dto/workout/ExerciseTypeResponse.java
package tom.projects.monkeylog.dto.workout;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Data;
import tom.projects.monkeylog.model.MetricFormat;
import tom.projects.monkeylog.model.exercise.ExerciseType;

@Data
@Builder
public class ExerciseTypeResponse {
    @NotEmpty
    private ExerciseType id;
    @NotEmpty
    private String name;
    @NotNull
    private MetricFormat metricFormat;
}

File: backendkt/src/main/kotlin/tom/com/monkeylog/dto/workout/ExerciseTypeResponse.kt
package tom.com.monkeylog.dto.workout

import tom.com.monkeylog.model.MetricFormat
import tom.com.monkeylog.model.exercise.ExerciseType

data class ExerciseTypeResponse(
    val id: ExerciseType,
    val name: String,
    val metricFormat: MetricFormat,
)
